solver/src/settlement/settlement_encoder.rs

```diff
@@ -15,12 +15,15 @@ use std::{collections::HashMap, iter};
 /// Additionally, the fact that the settlement is kept in an intermediate
 /// representation allows the encoder to potentially perform gas optimizations
 /// (e.g. collapsing two interactions into one equivalent one).
-#[derive(Debug)]
+#[derive(Debug, Clone)]
 pub struct SettlementEncoder {
     tokens: Vec<H160>,
     clearing_prices: HashMap<H160, U256>,
     trades: Vec<Trade>,
-    execution_plan: Vec<Box<dyn Interaction>>,
+    // This is an Arc so that this struct is Clone. Cannot require `Interaction: Clone` because it
+    // would make the trait not be object safe which prevents using it through `dyn`.
+    // TODO: Can we fix this in a better way?
```

Comment:
Not  going to be possible once we do merging because I imagine we'll want the option to still settle individual settlements if the merged one fails to simulate.